src/sagemaker/model_card/model_card.py

```diff
@@ -208,6 +208,257 @@ def search_model_associated_model_cards(model_id: str):
+                model_package_name (str, optional): Name of the model package (default: None),
+                model_package_version (int, optional): Version of the model package (default: None),
+                domain (str, optional): The machine learning domain of the model package you specified. Common machine learning domains include computer vision and natural language processing. (default: None),
+                task (str, optional): The machine learning task you specified that your model package accomplishes. Common machine learning tasks include object detection and image classification. (default: None),
+                created_by (ModelPackageCreator, optional): Information about the user who created model package. (default: None),
+                source_algorithms (SourceAlgorithms, optional):A list of algorithms that were used to create a model package. (default: None),
+                inference_specification (InferenceSpecification, optional): Details about inference jobs that can be run with models based on this model package. (default: None)
+        """
+        self.model_package_arn = model_package_arn
+        self.model_package_description = model_package_description
+        self.model_package_status = model_package_status
+        self.model_approval_status = model_approval_status
+        self.approval_description = approval_description
+        self.model_package_group_name = model_package_group_name
+        self.model_package_name = model_package_name
+        self.model_package_version = model_package_version
+        self.domain = domain
+        self.task = task
+        self.created_by = created_by
+        self.source_algorithms = source_algorithms
+        self.inference_specification = inference_specification
+        
+    
+    @classmethod
+    def from_model_package_name(cls, model_package_name: str, sagemaker_session: Session = None, **kwargs):
+        """Initialize a model package details from auto-discovered data.
+
+        Args:
+            model_package_name (str): The ARN of the model package version.
+            sagemaker_session (Session, optional): A SageMaker Session
+                object, used for SageMaker interactions (default: None). If not
+                specified, a SageMaker Session is created using the default AWS configuration
+                chain.
+            **kwargs: Other arguments in ModelPackage, i.e. model_package_description,
+                model_package_status, model_approval_status, model_approval_description, model_package_group_name, domain, task,  model_package_version, 
+                algorithm_specification, model_package_creator, inference_specification
+        Raises:
+            ValueError: A model with this name does not exist.
+            ValueError: A model card already exists for this model.
+        """
+
+        OPTIONAL_FIELDS_MAP = {
```

Comment:
move to the class level. We don't want to reinitiate this variable every time when we call from_model_package_name